plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -5,31 +5,36 @@ import (
 	"errors"
 	"fmt"
 	"log"
+	"math"
 	"strings"
 	"sync"
 	"time"
 
 	"github.com/Azure/go-autorest/autorest/adal"
 	mssql "github.com/denisenkom/go-mssqldb"
 	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
 	"github.com/influxdata/telegraf/filter"
 	"github.com/influxdata/telegraf/plugins/inputs"
 )
 
 // SQLServer struct
 type SQLServer struct {
-	Servers      []string `toml:"servers"`
-	AuthMethod   string   `toml:"auth_method"`
-	QueryVersion int      `toml:"query_version"`
-	AzureDB      bool     `toml:"azuredb"`
-	DatabaseType string   `toml:"database_type"`
-	IncludeQuery []string `toml:"include_query"`
-	ExcludeQuery []string `toml:"exclude_query"`
-	HealthMetric bool     `toml:"health_metric"`
-	pools        []*sql.DB
-	queries      MapQuery
-	adalToken    *adal.Token
-	muCacheLock  sync.RWMutex
+	Servers                 []string        `toml:"servers"`
+	AuthMethod              string          `toml:"auth_method"`
+	QueryVersion            int             `toml:"query_version"`
+	AzureDB                 bool            `toml:"azuredb"`
+	DatabaseType            string          `toml:"database_type"`
+	IncludeQuery            []string        `toml:"include_query"`
+	ExcludeQuery            []string        `toml:"exclude_query"`
+	HealthMetric            bool            `toml:"health_metric"`
+	RetryCount              int             `toml:"retry_count"`
+	RetryWaitTime           config.Duration `toml:"retry_wait_time"`
+	RetryExponentialBackoff bool            `toml:"retry_exponential_backoff"`
+	pools                   []*sql.DB
```

Comment:
Please use the telegraf logging facility for new loggings
```suggestion
	Log                     telegraf.Logger `toml:"-"`
	pools                   []*sql.DB
```